packages/transaction-controller/src/TransactionController.ts

```diff
@@ -610,9 +619,94 @@ export class TransactionController extends BaseControllerV1<
       this.onBootCleanup();
     });
 
+    this.onNetworkStateChange = onNetworkStateChange;
+    this.isMultichainEnabled = isMultichainEnabled;
+    this.incomingTransactions = incomingTransactions;
+    this.getNetworkClientRegistry = getNetworkClientRegistry;
+
     this.onBootCleanup();
   }
 
+  /**
+   * Initializes `registry`, `nonceTracker`, `#multichainTrackingHelper`,
+   * `incomingTransactionHelper`, `pendingTransactionTracker`, and passes a
+   * callback to `onNetworkStateChange`.
+   *
+   * This relies on the provider and block tracker being defined which may not
+   * have been the case at the time the Transaction Controller was instantiated.
+   *
+   * @param passedProvider - Reference to the provider proxy object
+   * @param passedBlockTracker - Reference to the block tracker proxy object
+   */
+  delayedInit(passedProvider: Provider, passedBlockTracker: BlockTracker) {
```

Comment:
If this is solely to support the global network `blockTracker` and `provider` instances, then this should no longer be an issue until this controller goes full multi-chain as these options will no longer exist since all the provider and block tracker usage will be dynamic based on the `networkClientId` provided when adding transactions.